scripts/test-tls13-eddsa-in-certificate-verify.py

```diff
@@ -0,0 +1,610 @@
+    print("                it will be reported in the test summary and the whole script will fail.")
+    print("                May be specified multiple times.")
+    print(" -X message     expect the `message` substring in exception raised during")
+    print("                execution of preceding expected failure probe")
+    print("                usage: [-x probe-name] [-X exception], order is compulsory!")
+    print(" -n num         run 'num' or all(if 0) tests instead of default(10)")
+    print("                (excluding \"sanity\" tests)")
+    print(" -s sigalgs     hash and signature algorithm pairs that the server")
+    print("                is expected to support. Either pairs of algorithms")
+    print("                (\"sha1+ecdsa\"), pairs of identifiers (\"1+1\")")
+    print("                or the TLS 1.3 names (\"rsa_pss_rsae_sha256\").")
+    print("                Multiple values separated by spaces.")
+    print(" --hash-order   the order in which hashes are preferred in some")
+    print("                tests, can be used to test different combinations")
+    print(" -k keyfile     file with private key of client")
+    print(" -c certfile    file with the certificate of client")
+    print(" --help         this message")
+
+
+def hashes_to_list(arg):
+    hlist = []
+    for h in arg.split():
+        name = None
+        hnum = getattr(HashAlgorithm, h, None)
+        if not hnum:
+            try:
+                hnum = int(h)
+            except ValueError:
+                pass
+        if hnum:
+            name = HashAlgorithm.toRepr(hnum)
+        if not name:
+            raise ValueError("Invalid Hash Id or Name {0}".format(h))
+        hlist.append(name)
+    if len(hlist) < 2:
+        raise ValueError(
+            "The ordered list of hashes must contain at least 2 elements," +
+            " found {0}".format(len(hlist)))
+    return hlist
+
+
+def sigalg_select(alg_type, hash_pref, supported=None, cert_type=None):
+    for hash_name in hash_pref:
+        if not cert_type:
+            name = "_".join([alg_type, hash_name])
+        elif cert_type == "rsa":
+            name = "_".join([alg_type, "rsae", hash_name])
+        elif cert_type == "rsa-pss":
+            name = "_".join([alg_type, "pss", hash_name])
+        elif cert_type == "ecdsa":
+            name = "_".join([alg_type, hash_name])
+        else:
+            raise ValueError("Unknown certificate type {0}".format(cert_type))
+
+        sigalg = getattr(SignatureScheme, name)
+
+        if supported is None:
+            return sigalg
+        if sigalg in supported:
+            return sigalg
+
+    raise ValueError(
+        "Couldn't find a supported Signature Algorithm that  matches the" +
+        " provided parameters: {0}, {1}, {3}".format(alg_type, hash_pref,
+                                                    cert_type))
```

Comment:
I don't see this function used in the script.

Reply:
yes, leftover from copy-paste, fixed